panel/models/data.ts

```diff
@@ -6,10 +6,29 @@ export function transform_cds_to_records(cds: ColumnDataSource, addId: boolean =
   const cdsLength = cds.get_length()
   if (columns.length === 0||cdsLength === null)
     return []
+
+  // Unpack typed arrays in a array
+  let total_length: number = 0
+  let arrays: any = {}
   for (let i = start; i < cdsLength; i++) {
+    for (const column of columns) {
+      arrays[column] = []
+      let unpack_array: any = cds.get_array(column);
+      for (let i = start; i < unpack_array.length; i++) {
+        if ((unpack_array[i].byteLength !== undefined) && unpack_array[i].length > 1)
+          // A typed array
+          arrays[column].push.apply(arrays[column], Array(...unpack_array[i]));
+        else
+          arrays[column].push(unpack_array[i])
+        }
+      total_length = Math.max(total_length, arrays[column].length);
+      }
+    }
```

Comment:
Can you explain the issue? This looks like a workaround for a bokeh bug.

Reply:
I would classify it as a workaround, but I only see this problem on `pn.widgets.Tabulator` and not with `pn.widgets.DataFrame`. 

The problem is that when streaming data, the data sent is a typed array and not an `Array`, e.g., the following code and a `console.log(data)` at the end of the function `transform_cds_to_records`, will give the following output, when clicking on three times on the button:

![image](https://user-images.githubusercontent.com/19758978/217230640-da79545c-cd4f-4b35-8d7e-2d9fef8b2ee0.png)

![image](https://user-images.githubusercontent.com/19758978/217230510-e3ae59bc-b7ab-4344-b88f-bcc27c3d712d.png)

This is the code I used:

``` python
import numpy as np
import pandas as pd
import panel as pn

df = pd.DataFrame(np.random.random((3, 2)), columns=["A", "B"])
widget = pn.widgets.Tabulator(df).servable()


def stream_data(*e) -> None:
    widget.stream(df)  


b = pn.widgets.Button(name="Test").servable()
b.on_click(stream_data)
```

Reply:
I have added a comment about it in the code. Will remove the workarounds when the upstream issue gets fixed: https://github.com/bokeh/bokeh/issues/12788

Reply:
Would suggest removing it now and marking the streaming tests as xfail for the time being, the workaround breaks more tests than it fixes and I'd like to merge this PR asap.

Reply:
Will do

Reply:
Without removing the data workaround, no tabulator tests are failing. Would you still like for me to remove it?

The problem workaround was fixed in 1fab8bd 

```
# Unit tests
FAILED panel/tests/test_server.py::test_server_exception_handler_async_change_event[threads-synchronous_handler] - assert 0 == 1

# Examples
examples/reference/widgets/DatePicker.ipynb

# UI Test
FAILED panel/tests/ui/io/test_jupyter_server_extension.py::test_jupyter_server - AssertionError: assert '' == '0'
FAILED panel/tests/ui/io/test_jupyter_server_extension.py::test_jupyter_server_custom_resources - AssertionError: assert 'rgb(0, 0, 0)' == 'rgb(128, 0, 128)'
FAILED panel/tests/ui/io/test_jupyter_server_extension.py::test_jupyter_server_kernel_error - TimeoutError: wait_until timed out in 5000 milliseconds
FAILED panel/tests/ui/pane/test_markup.py::test_update_markdown_pane_resizes - assert 63.125 == 64
FAILED panel/tests/ui/pane/test_plotly.py::test_plotly_hover_data - AssertionError: assert None == {'points': [{'curveNumber': 0, 'pointIndex': 0, 'pointNumber': 0, 'x': 0, ...}]}
FAILED panel/tests/ui/pane/test_plotly.py::test_plotly_click_data - AssertionError: assert None == {'points': [{'curveNumber': 0, 'pointIndex': 0, 'pointNumber': 0, 'x': 0, ...}]}
FAILED panel/tests/ui/template/test_fastgridtemplate.py::test_fast_grid_template_updates - playwright._impl._api_types.TimeoutError: Timeout 20000ms exceeded.
FAILED panel/tests/ui/template/test_materialtemplate.py::test_material_template_no_console_errors - assert [<ConsoleMess... (Not Found)>] == []
FAILED panel/tests/ui/widgets/test_image.py::test_pdf_no_embed_start_page - playwright._impl._api_types.TimeoutError: Timeout 20000ms exceeded.
FAILED panel/tests/ui/widgets/test_sliders.py::test_editableslider_textinput_end[EditableFloatSlider] - playwright._impl._api_types.TimeoutError: Timeout 20000ms exceeded.
FAILED panel/tests/ui/widgets/test_sliders.py::test_editableslider_textinput_start[EditableFloatSlider] - playwright._impl._api_types.TimeoutError: Timeout 20000ms exceeded.
FAILED panel/tests/ui/widgets/test_sliders.py::test_editablerangeslider_textinput_end[EditableRangeSlider] - playwright._impl._api_types.TimeoutError: Timeout 20000ms exceeded.
FAILED panel/tests/ui/widgets/test_sliders.py::test_editablerangeslider_textinput_start[EditableRangeSlider] - playwright._impl._api_types.TimeoutError: Timeout 20000ms exceeded.
```

Reply:
Okay, let's leave it for now then.